middlewares/logResponse.js

```diff
@@ -1,8 +1,57 @@
+const Sentry = require('@prairielearn/sentry');
+
+const config = require('../lib/config');
 const logger = require('../lib/logger');
 
 module.exports = function (req, res, next) {
   if (req.method !== 'OPTIONS') {
-    res.on('finish', function () {
+    res.once('finish', function () {
+      // Attach information to the Sentry scope so that we can more easily
+      // debug errors. No PII is added.
+      if (config.sentryDsn) {
+        Sentry.configureScope((scope) => {
+          if (res.locals?.course?.id) {
+            scope.setContext('course', {
+              id: res.locals.course?.id,
+              name: res.locals.course?.name,
+            });
+          }
+
+          if (res.locals?.course_instance?.id) {
+            scope.setContext('course_instance', {
+              id: res.locals.course_instance?.id,
+              name: res.locals.course_instance?.name,
+            });
+          }
+
+          if (res.locals?.assessment?.id) {
+            scope.setContext('assessment', {
+              id: res.locals.assessment?.id,
+              directory: res.locals.assessment?.directory,
+            });
+          }
+
+          if (res.locals?.assessment_instance?.id) {
+            scope.setContext('assessment_instance', {
+              id: res.locals.assessment_instance?.id,
+            });
+          }
+
+          if (res.locals?.question?.id) {
+            scope.setContext('question', {
+              id: res.locals.question?.id,
+              directory: res.locals.question?.directory,
+            });
+          }
+
+          if (res.locals?.instance_question?.id) {
+            scope.setContext('instance_question', {
+              id: res.locals.instance_question?.id,
+            });
+          }
+        });
+      }
```

Comment:
Could we also add `res.locals.response_id`, `req.method`, `req.path`, and things like that?

What about `user_id`, or are we considering that to be PII?

Reply:
Ah yes, I forgot the one thing you asked about 😂 I'll add `response_id`.

The method and path are already captured automagically by Sentry:

<img width="745" alt="Screen Shot 2022-08-31 at 16 22 27" src="https://user-images.githubusercontent.com/1476544/187801845-8eab409d-40bc-4afc-8b39-158cd7abcacc.png">

Reply:
`user_id` is captured elsewhere, although I should move it to this file for consistency.

Reply:
> Ah yes, I forgot the one thing you asked about 😂 I'll add `response_id`.

🙏 

> The method and path are already captured automagically by Sentry:

Nice!